backend/v3/storage/database/dialect/postgres/migration/002_organization_table/up.sql

```diff
@@ -0,0 +1,30 @@
+CREATE TYPE zitadel.organization_state AS ENUM (
+	'active',
+	'inactive'
+);
+
+CREATE TABLE zitadel.organizations(
+  id TEXT NOT NULL CHECK (id <> ''),
+  name TEXT NOT NULL CHECK (name <> ''),
+  instance_id TEXT NOT NULL CHECK (instance_id <> '') REFERENCES zitadel.instances (id),
+  state zitadel.organization_state NOT NULL,
+  created_at TIMESTAMPTZ DEFAULT NOW() NOT NULL,
+  updated_at TIMESTAMPTZ DEFAULT NOW() NOT NULL,
+  deleted_at TIMESTAMPTZ DEFAULT NULL,
+
+  PRIMARY KEY (instance_id, id),
+  UNIQUE      (instance_id, name)
+);
```

Comment:
names should only be unique on not deleted organizations
```suggestion
  PRIMARY KEY (instance_id, id),
  UNIQUE      (instance_id, name) WHERE deleted_at IS NULL
);
```